WordPress/Classes/Services/PostService.m

```diff
@@ -593,6 +389,82 @@ - (AbstractPost *)findPostWithID:(NSNumber *)postID inBlog:(Blog *)blog {
     return [posts firstObject];
 }
 
+- (NSDictionary *)remoteSyncParametersDictionaryForRemote:(nonnull id <PostServiceRemote>)remote
```

Comment:
Added a commit with `const` for all the keys and values. I thought a bit about moving this constructor method to the remote classes but I think it should stay for the moment. Most of the parameters are the same, and only a couple aren't supported at all within XML-RPC.

Also, in the current context of `PostService` it seems to make most sense for it to be aware of what the remotes can handle and what they cannot. Perhaps if XML-RPC supported everything REST did, we would just have a key/value translator for each remote. Otherwise, `PostService` doesn't necessarily _know_ that XML-RPC can't handle a search.